apiserver/facades/client/application/backend.go

```diff
@@ -313,6 +315,10 @@ func (s stateShim) AddApplication(args state.AddApplicationArgs) (Application, e
 	return stateApplicationShim{a, s.State}, nil
 }
 
+func (s stateShim) AddPendingResource(appName string, chRes resource.Resource) (string, error) {
+	return s.State.Resources().AddPendingResource(appName, "", chRes)
```

Comment:
The AddPendingResource call should include the userID rather that shimming the method to not pass the data.

Reply:
Yeah I realized there's something going on there while I was writing it, looks like in the current code we're doing the same thing too https://github.com/juju/juju/blob/71a9532916556e8cd009459372d18da4effcaf77/apiserver/facades/client/resources/facade.go#L214

Any quick ideas about where to get that info from? couldn't find it in the first glance

Reply:
It looks like we don't include the username here with add pending resource. I had to deploy a charm and use a resource to figure it out. Not good.

```
{
	"_id" : "3ba0dee1-b1fc-4748-802a-cdd6d02b4a44:resource#juju-qa-test/foo-file#unit-juju-qa-test/0",
	"resource-id" : "juju-qa-test/foo-file",
	"pending-id" : "",
	"application-id" : "juju-qa-test",
	"unit-id" : "juju-qa-test/0",
	"name" : "foo-file",
	"type" : "file",
	"path" : "foo.txt",
	"description" : "foo resource.",
	"origin" : "store",
	"revision" : 2,
...
	"username" : "juju-qa-test/0",
...
}
```